### ISSUE TITLE
OpenTracing Registry Page

### ISSUE THREAD

**Issue by austinlparker** [opentracing/opentracing.io#343]
This is the tracking issue for the OpenTracing Registry, a page that's intended to be a searchable index of tracers, addons, plugins, and client libraries that support OpenTracing.

The intention of this registry is to improve the discoverability of OpenTracing components both inside and outside the opentracing and opentracing-contrib organizations on GitHub.

**Comment by austinlparker** [comment]
Yeah, pretty much. I've already got some initial work here https://github.com/opentracing/opentracing.io/compare/feature/searchableRegistry if you want to see.

The biggest difference is that instead of a single file listing modules, we'll have a bunch of markdown files that get turned into JSON and then searched. Should make it easy for people to do PR's of new modules.

**Comment by austinlparker** [comment]
@tedsuo @yurishkuro I'm trying to come up with a set of standard `registryType`s for things that can go into the registry; so far I've got 'tracer' and 'instrumentation'. I'm not really sold on either of those names or what they refer to right now, however. Y'all have any thoughts? This is a WIP screenshot of how those will be displayed.

![image](https://user-images.githubusercontent.com/4140740/46830516-d8ab4600-cd6e-11e8-9462-0713e900a93b.png)

**Comment by yurishkuro** [comment]
tracer / instrumentation labels make sense to me.

I would suggest (without thinking through implementation implications) keeping the individual files in two different directories, tracers & instrumentations, so that people won't need to remember to tag them accordingly.

Another useful tag is language: Java, Go, etc.

**Comment by tedsuo** [comment]
I'm not sure if we want to have explicit categories  (language, type, etc) and then some additional generic tags, or just tags.

In other words: is metadata `key:value` like categories, or just `value` like labels?

**Comment by austinlparker** [comment]
Here's a WIP gif of where I'm at with the search - I'm thinking having a single list page that has all of the items on it that you can filter/search down feels pretty good, but I'm pretty open to suggestions still.

![2018-10-12 16 13 24](https://user-images.githubusercontent.com/4140740/46892010-ef6a9f00-ce39-11e8-9c37-2d647f23e671.gif)

One thing I don't really care for so far is that I'm doing a POST each time you search, so you get that annoying redraw since we're reloading the page. That said, the nice thing about doing a POST is that you can share a link with a search query (i.e., `https://opentracing.io/registry/?s=Java`)

**Comment by austinlparker** [comment]
@tedsuo right now metadata tags are a value array, and the type of thing is a special `k:v` (`registryType` in the below snippet, because `type` is reserved in Hugo front matter)

```---
title: OpenTracing JDBI Instrumentation
registryType: instrumentation
tags:
  - java
  - opentracing
  - jdbi
  - database
repo: https://github.com/opentracing-contrib/java-jdbi
license: Apache 2.0
description: OpenTracing instrumentation for JDBI
authors: OpenTracing Contributors
---
```

**Comment by tedsuo** [comment]
yeah, in general this is great.. I might suggest that `registryType` just be a tag, so that everything is a tag. That give us flexibility (some projects may span multiple languages, so projects may be both an instrumentation lib and a tracer).
